### PR TITLE
#157 Admin Registration Flow Bugfix

### PR DESCRIPTION
**Pull request by moaydogdu**
# PULL REQUEST TEMPLATE

## Description

The problem of using the Verification ID in the Admin Registration Flow more than once.

## Related Issue

#157 "Admin Kayıt Akışındaki Verification ID'nin Birden Fazla Kez Kullanılması"

## Proposed Changes

The problem was solved by throwing an "AysAdminUserRegisterVerificationCodeNotValidException" error when the "verification id" value used during the creation of a new admin is not "AdminUserRegisterVerificationStatus.WAITING". At the end of the related flow, the status of the "AdminUserRegisterVerificationEntity" object associated with the "verification id" is already changed to "AdminUserRegisterVerificationStatus.COMPLETED".

In this way, when the request is sent again with the same verification id value, the "AysAdminUserRegisterVerificationCodeNotValidException" error is thrown because it is "AdminUserRegisterVerificationStatus.COMPLETED" and reuse is prevented.

## Checklist

Please check the following before submitting your pull request:

- [X] I have tested my changes locally and they are working as expected.
- [X] The code is formatted according to the project's coding guidelines and style.
- [X] Necessary documentation has been added or existing documentation has been updated.
- [X] Relevant unit tests have been written and included.
- [X] Relevant integration tests have been written and included.
- [X] The code has been reviewed to ensure its quality.
- [X] The code does not contain any issues flagged by SonarLint.
- [X] Default reviewers have been assigned to this pull request.
- [X] Assignees have been added if necessary.
- [X] Labels have been applied if necessary.
- [X] If an issue is opened and this pull request is related to that issue, it should not be associated with any project
  board or milestone. However, if there is no issue, the pull request should be associated with the appropriate project
  board (BE Board, In Review column) and milestone to track the progress.

## Reviewers

@agitrubard @skayikci @Rapter1990

### CODE REVIEW

File: src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java [hunk 1]
```diff
@@ -56,10 +57,13 @@ void givenValidAdminUserRegisterRequest_thenAdminUserRegistered() {
                 .withEmail(AysTestData.VALID_EMAIL)
                 .withPhoneNumber(mockPhoneNumber).build();
 
+
         // When
-        AdminUserRegisterVerificationEntity mockAdminUserRegisterVerificationEntity = new AdminUserRegisterVerificationEntityBuilder().build();
+        AdminUserRegisterVerificationEntity mockAdminUserRegisterVerificationEntity = new AdminUserRegisterVerificationEntityBuilder()
+                .withValidFields()
+                .build();
         Mockito.when(adminUserRegisterVerificationRepository.findById(Mockito.anyString()))
```
Comment (Rapter1990) [inline src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java:65]: `AdminUserRegisterVerificationEntity` `when`'den önce kullanabilir miyiz. When'den sonra `Mockito.when` kullanalım.

File: src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java [hunk 2]
```diff
@@ -130,6 +135,36 @@ void givenInvalidVerificationIdFromAdminUserRegisterRequest_whenVerificationEnti
...
+                .withInstitutionId(AysRandomUtil.generateUUID())
+                .withEmail(AysTestData.VALID_EMAIL)
+                .withPhoneNumber(mockPhoneNumber).build();
+
+        // When
+        AdminUserRegisterVerificationEntity mockAdminUserRegisterVerificationEntity =
+                new AdminUserRegisterVerificationEntityBuilder()
+                        .withStatus(AdminUserRegisterVerificationStatus.COMPLETED)
+                        .build();
+
```
Comment (Rapter1990) [inline src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java:154]: `AdminUserRegisterVerificationEntity` `when`'den önce kullanabilir miyiz. When'den sonra `Mockito.when` kullanalım.

File: src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java [hunk 3]
```diff
@@ -139,7 +174,9 @@ void givenInvalidInstitutionIdFromAdminUserRegisterRequest_whenInstitutionNotExi
                 .withInstitutionId("Invalid").build();
 
         // When
-        AdminUserRegisterVerificationEntity mockAdminUserRegisterVerificationEntity = new AdminUserRegisterVerificationEntityBuilder().build();
```
Comment (Rapter1990) [inline src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java:142]: `AdminUserRegisterVerificationEntity` `when`'den önce kullanabilir miyiz. When'den sonra `Mockito.when` kullanalım.

File: src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java [hunk 4]
```diff
@@ -168,7 +205,9 @@ void givenExistingEmailFromAdminUserRegisterRequest_whenAdminUserExist_thenThrow
                 .withEmail(AysTestData.VALID_EMAIL).build();
 
         // When
-        AdminUserRegisterVerificationEntity mockAdminUserRegisterVerificationEntity = new AdminUserRegisterVerificationEntityBuilder().build();
```
Comment (Rapter1990) [inline src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java:171]: `AdminUserRegisterVerificationEntity` `when`'den önce kullanabilir miyiz. When'den sonra `Mockito.when` kullanalım.

File: src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java [hunk 5]
```diff
@@ -202,7 +241,9 @@ void givenExistingUsernameFromAdminUserRegisterRequest_whenAdminUserExist_thenTh
...
```
Comment (Rapter1990) [inline src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java:205]: `AdminUserRegisterVerificationEntity` `when`'den önce kullanabilir miyiz. When'den sonra `Mockito.when` kullanalım.

File: src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java [hunk 6]
```diff
@@ -243,7 +284,9 @@ void givenExistingPhoneNumberFromAdminUserRegisterRequest_whenAdminUserExist_the
                 .withPhoneNumber(mockPhoneNumber).build();
 
         // When
-        AdminUserRegisterVerificationEntity mockAdminUserRegisterVerificationEntity = new AdminUserRegisterVerificationEntityBuilder().build();
```
Comment (Rapter1990) [inline src/test/java/com/ays/admin_user/service/impl/AdminUserRegisterServiceImplTest.java:246]: `AdminUserRegisterVerificationEntity` `when`'den önce kullanabilir miyiz. When'den sonra `Mockito.when` kullanalım.

**Review by Rapter1990 — APPROVED** [review]
LGTM!

**Review by AhmetAksunger — APPROVED** [review]
LGTM!

**Review by agitrubard — APPROVED** [review]

**Pull request merged**